src/ol/format/wkt.js

```diff
@@ -168,16 +169,28 @@ ol.format.WKT.encode_ = function(geom) {
   ol.DEBUG && console.assert(geometryEncoder, 'geometryEncoder should be defined');
   var enc = geometryEncoder(geom);
   type = type.toUpperCase();
+  if (geom.getLayout) {
+    var layout = geom.getLayout();
+    var dimInfo = '';
+    if (layout === ol.geom.GeometryLayout.XYZ || layout === ol.geom.GeometryLayout.XYZM) {
+      dimInfo += 'Z';
+    }
+    if (layout === ol.geom.GeometryLayout.XYM || layout === ol.geom.GeometryLayout.XYZM) {
+      dimInfo += 'M';
+    }
+    if (dimInfo.length > 0) {
+      type += ' ' + dimInfo;
+    }
+  }
   if (enc.length === 0) {
     return type + ' ' + ol.format.WKT.EMPTY;
   }
   return type + '(' + enc + ')';
 };
 
-
 /**
  * @const
- * @type {Object.<string, function(ol.geom.Geometry): string>}
+ * @type {Object.<string, function((ol.geom.SimpleGeometry|ol.geom.GeometryCollection)): string>}
```

Comment:
It was needed at some point. Perhaps before the rewrite to use GeometryLayout.
I believe it's more correct since it describes the limitations better. But I can change back to Geometry now.